ReportingServicesTools/Functions/CatalogItems/Update-RsSubscription.ps1

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2016 Microsoft Corporation. All Rights Reserved.
+# Licensed under the MIT License (MIT)
+
+
+function Update-RsSubscription
+{
+  <#
+        .SYNOPSIS
+            This script will update subscriptions piped from get-RsSubscriptions
+
+        .DESCRIPTION
+            This script will take the custom object producted by get-RSSubscription and use the data to update the 
+            matchdata xml to either a new startdatetime, or enddate.
+
+        .PARAMETER ReportServerUri
+            Specify the Report Server URL to your SQL Server Reporting Services Instance.
+            Use the "Connect-RsReportServer" function to set/update a default value.
+
+        .PARAMETER Credential
+            Specify the password to use when connecting to your SQL Server Reporting Services Instance.
+            Use the "Connect-RsReportServer" function to set/update a default value.
+
+        .PARAMETER Proxy
+            Report server proxy to use.
+            Use "New-RsWebServiceProxy" to generate a proxy object for reuse.
+            Useful when repeatedly having to connect to multiple different Report Server.
+
+        .EXAMPLE
+            $proxy = "http://ssrs-t-1234/Reportserver"
+            Get-RsSubscription -ReportServerUri $proxy -path '/Finance/ImportantReports' | 
+            where description -eq 'super important' | Update-RsSubscription -ReportServerUri $proxy -EndDate 9/9/2099
+
+            assign proxy to variable for reuse.
+            use get-rsssubscripion to find the 'super important' report in my /finance/importantreports path. Then Pipe the results
+            to update-RsSubscription where we reuse the same $proxy to update the EndDate to 9/9/2099.
+
+        .EXAMPLE
+            $proxy = "http://ssrs-t-1234/Reportserver"
+
+            [psobject]$subscriptions = Get-RsSubscription -ReportServerUri $proxy -path '/Finance/ImportantReports' | where description -eq 'super important' 
+            $subscriptions | Update-RsSubscription -ReportServerUri $proxy -EndDate 9/9/2099 -StartDate 1/1/2016
+
+            assign proxy to variable for reuse. 
+            Then create a psobject variable to store the objects created from get-RSSubscription.
+            Then Pipe the results to Update-RSSubscription where we reuse the same $proxy to update the EndDate to 9/9/2999 and StartDate to 1/1/2016
+        
+    #>
+    
+  [CmdletBinding()]
+  param (
+    [string]
+    $ReportServerUri,
+    [Alias('ReportServerCredentials')]
+    [System.Management.Automation.PSCredential]
+    $Credential,
+    $Proxy,
+    [parameter(Mandatory = $false)]
+    [DateTime]$StartDateTime,
+    [parameter(Mandatory = $false)]
+    [DateTime]$EndDate,
+    [Parameter(Mandatory = $true, ValueFromPipeLine)]
+    [PSCustomObject[]]$SubProperties

```

Comment:
I am not really sure whether this needs to be specified or not when calling this cmdlet. Looking at the examples, it seems like any property that is not one of the other properties listed will be assigned to this array. If that is the case, I would recommend you to replace this to more explicit parameter names (e.g. StartDate, EndDate, Description, etc.) so that the PowerShell intellisense will help them when they are invoking this cmdlet.

Reply:
Since Get-RsSubscription returns a custom object, Update-RsSubscription only takes a custom object so it can see it from the pipeline.